Source/WebCore/loader/DocumentLoader.cpp

```diff
@@ -2684,6 +2696,21 @@ Ref<CachedResourceLoader> DocumentLoader::protectedCachedResourceLoader() const
     return m_cachedResourceLoader;
 }
 
+void DocumentLoader::whenDocumentIsCreated(Function<void(Document*)>&& callback)
+{
+    RELEASE_ASSERT(!m_canUseServiceWorkers || !!m_resultingClientId);
+
+    if (m_whenDocumentIsCreatedCallback) {
+        auto previousCallback = std::exchange(m_whenDocumentIsCreatedCallback, { });
+        m_whenDocumentIsCreatedCallback = [previousCallback = WTFMove(previousCallback), callback = WTFMove(callback)] (auto* document) mutable {
+            previousCallback(document);
```

Comment:
I'm having a bit of a brain freeze here but how does this not require a `if (!previousCallback)` guard?

Reply:
We are checking `m_whenDocumentIsCreatedCallback` first.
If not null, we are moving it to `previousCallback` via std::exchange, so `previousCallback` should not be null.
I wonder how I can make this easier to read.

Reply:
I made some changes to make it clearer that previousCallback is not null, and I renamed callback to newCallback in the lambda.